test/test_vmap.py

```diff
@@ -1988,6 +1988,37 @@ def test_slice(self, device):
         self._batched_grad_test(lambda x: x[:, 1:3], (x,), {})
         self._batched_grad_test(lambda x: x[..., 1:3], (x,), {})
 
+    @allowVmapFallbackUsage
+    def test_inplace_view(self, device):
+        leaf = torch.randn(4, 5, requires_grad=True)
+
+        def func(leaf):
+            # Make sure the function is non-trivially twice differentiable
+            base = leaf * leaf
+            view = base[0]
+            view.cos_()
+            return view
+
+        self._batched_grad_test(func, (leaf,), {})
+        self._batched_grad_grad_test(func, (leaf,), {})
+
+    @allowVmapFallbackUsage
+    def test_inplace_manyview(self, device):
+        leaf = torch.randn(4, 4, 5, requires_grad=True)
+
+        def func(leaf):
+            # Make sure the function is non-trivially twice differentiable
+            base = leaf * leaf
+            view = base.transpose(0, 2)
+            view = view[1]
+            view = base.diagonal()
```

Comment:
Not sure what is the purpose of the first two lines as they get overwritten here.

Reply:
Oops.

Reply:
Fixed; thanks for the catch!